packages/formspree-core/src/core.ts

```diff
@@ -1,40 +1,37 @@
 import type { Stripe } from '@stripe/stripe-js';
-import type {
-  SubmissionData,
-  SubmissionOptions,
-  SubmissionBody,
-  SubmissionResponse,
-} from './forms';
-import {
-  appendExtraData,
-  clientHeader,
-  encode64,
-  handleLegacyErrorPayload,
-  handleSCA,
-} from './utils';
 import { Session } from './session';
+import {
+  SubmissionErrorResult,
+  SubmissionRedirectResult,
+  SubmissionStripePluginPendingResult,
+  isServerErrorResponse,
+  isServerRedirectResponse,
+  isServerStripePluginPendingResponse,
+  type FieldValues,
+  type SubmissionData,
+  type SubmissionOptions,
+  type SubmissionResult,
+} from './submission';
+import { appendExtraData, clientHeader, encode64 } from './utils';
 
 export interface Config {
+  fetch?: typeof window.fetch;
   project?: string;
   stripePromise?: Stripe;
 }
 
 export class Client {
+  private readonly fetch: typeof window.fetch;
   project: string | undefined;
   stripePromise: Stripe | undefined;
-  private session: Session | undefined;
+  private readonly session?: Session;
 
   constructor(config: Config = {}) {
+    this.fetch = config.fetch ?? window.fetch;
     this.project = config.project;
     this.stripePromise = config.stripePromise;
-    if (typeof window !== 'undefined') this.startBrowserSession();
-  }
 
-  /**
-   * Starts a browser session.
-   */
-  startBrowserSession(): void {
-    if (!this.session) {
```

Comment:
Clean this up & simplify.

All `startBrowserSession` does is to instantiate `Session` and set it to `this.session`. It is not meant to be called manually by users (and doing so it's actually a mistake). It was introduced to make it safe to create client on the server; in this PR, we change to instantiate the session only if the window exists, which happens on the client.

Removing it reduce confusion and getting rid of an unintended API.